pkg/start/start.go

```diff
@@ -132,15 +132,19 @@ func (o *Options) Run(ctx context.Context) error {
 	}
 	// check to see if techpreview should be on or off.  If we cannot read the featuregate for any reason, it is assumed
 	// to be off.  If this value changes, the binary will shutdown and expect the pod lifecycle to restart it.
-	includeTechPreview := false
+	startingFeatureSet := ""
 	gate, err := cb.ClientOrDie("feature-gate-getter").ConfigV1().FeatureGates().Get(ctx, "cluster", metav1.GetOptions{})
 	switch {
 	case apierrors.IsNotFound(err):
-		includeTechPreview = false // if we have no featuregates, then we aren't tech preview
+		// if we have no featuregates, then we assume the default featureset, which is "".
+		// This excludes everything that could possibly depend on a different feature set.
+		startingFeatureSet = ""
 	case err != nil:
 		klog.Warningf("Error getting featuregate value: %v", err)
```

Comment:
Hmm, we may need a pointer for passing this thing around (luckily, library-go is already using a pointer).  openshift/enhancements#1227 currently includes:

> If a special manifest is required for a FeatureSet that allows changes (LatencySensitive to Default for instance), the person adding the manifest will have to add the appropriate delete-annotated manifest in Default.  This is expected to be a rare occurrence of a rare occurrence.

So there's a risk here of:

1. User configures LatencySensitive.
2. Resource A, that only happens in LatencySensitive, rolls out.
3. Happy days.
4. Something bumps the CVO.
5. New CVO comes up, but network hiccup fails the FeatureGate GET.
6. We enter this `err != nil` case, and warn, but leave `startingFeatureSet == ""`.
7. Reconciling manifests, we exclude the LatencySensitive resource A manifest (meh), but include the deletion manifest (!), and remove resource A from the cluster.
8. Change-detector has a successful GET, notices the mistake, and shuts down the CVO.
9. Replacement CVO has a successful GET, realizes it's LatencySensitive, and rolls resource A back into the cluster.

That's not the end of the world, but I'd feel more comfortable with pointers tracking whether we knew or not about the feature gate, and... hmm, I guess there's not a clear way to tell library-go to exclude everything that declares a feature-set annotation.  Maybe we don't need a pointer, but we do need a dummy value here like:

```go
startingFeatureSet = "not sure, so exclude everything that might care"
```

?